crates/ol/chain-types/src/block_flags.rs

```diff
@@ -0,0 +1,32 @@
+//! Flags field for block header.
+
+use strata_codec_derive::Codec;
+
+type RawBlockFlags = u16;
+
+const IS_TERMINAL: RawBlockFlags = 0x0001;
+
+/// Flags in the block header that we use for various signalling purposes.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Codec)]
+pub struct BlockFlags(RawBlockFlags);
+
+impl BlockFlags {
+    /// Constructs a zero flag.
+    pub fn zero() -> Self {
+        Self(0)
+    }
+
+    /// Assigns the `IS_TERMINAL` flag to some value.
+    pub fn set_is_terminal(&mut self, b: bool) {
+        if b {
+            self.0 |= IS_TERMINAL;
+        } else {
+            self.0 &= !IS_TERMINAL;
+        }
+    }
```

Comment:
What is this magic trick?

Reply:
We define the consts as bitmasks so we can use bithacks to manipulate the flags field

```
aaaaaaaa | 00000001 = aaaaaaa1
aaaaaaaa & 11111110 = aaaaaaa0
```

Calling it a bithack is maybe misleading since this is literally just the most sensical way to do it that isn't absurdly verbose.  Idk why Rust uses `!` for bitwise inversion instead of `~` like most languages but that's just what the syntax is.